src/FormField/MultiLine.php

```diff
@@ -0,0 +1,764 @@
+     * Allow to Run expression base on rowData value.
+     */
+    private function setDummyModelValue($model)
+    {
+        $post = $_POST;
+
+        foreach ($this->fieldDefs as $def) {
+            $fieldName = $def['field'];
+            if ($fieldName === $model->id_field) {
+                continue;
+            }
+
+            $value = isset($post[$fieldName]) ? $post[$fieldName] : null;
+            if ($model->getElement($fieldName)->isEditable()) {
+                try {
+                    $model[$fieldName] = $value;
+                } catch (ValidationException $e) {
+                    //bypass validation at this point.
+                }
+            }
+        }
+
+        return $model;
+    }
+
+    /**
+     * Return values associated to field expression.
+     *
+     * @param $m
+     *
+     * @throws \atk4\core\Exception
+     * @throws \atk4\data\Exception
+     *
+     * @return array
+     */
+    private function getExpressionValues($m)
+    {
+        $dummyFields = [];
+        $formatValues = [];
+
+        foreach ($this->getExpressionFields($m) as $k => $field) {
+            $dummyFields[$k]['name'] = $field->short_name;
+            $dummyFields[$k]['expr'] = $this->getDummyExpression($field, $m);
+        }
+
+        if (!empty($dummyFields)) {
+            $dummyModel = new Model($m->persistence, ['table' => $m->table]);
+            foreach ($dummyFields as $f) {
+                $dummyModel->addExpression($f['name'], ['expr'=>$f['expr'], 'type' => $m->getElement($f['name'])->type]);
+            }
+            $values = $dummyModel->loadAny()->get();
+            unset($values[$m->id_field]);
+
+            foreach ($values as $f => $value) {
+                $field = $m->getElement($f);
+                $formatValues[$f] = $this->app->ui_persistence->_typecastSaveField($field, $value);
+            }
+        }
+
+        return $formatValues;
+    }
+
+    /**
+     * Get all field expression in model.
+     * But only evaluate expression used in rowFields.
+     *
+     * @return array
+     */
+    private function getExpressionFields($model)
+    {
+        $fields = [];
+        foreach ($model->elements as $f) {
+            if (!$f instanceof Field_SQL_Expression || !in_array($f->short_name, $this->rowFields)) {
+                continue;
+            }
+
+            $fields[] = $f;
+        }
+
+        return $fields;
+    }
+
+    /**
+     * Return expression where field are replace with their current or default value.
+     * ex: total field expression = [qty] * [price] will return 4 * 100
+     * where qty and price current value are 4 and 100 respectively.
+     *
+     * @param $expr
+     *
+     * @throws \atk4\core\Exception
+     *
+     * @return mixed
+     */
+    private function getDummyExpression($exprField, $model)
+    {
+        $expr = $exprField->expr;
+        $matches = [];
+
+        preg_match_all('/\[[a-z0-9_]*\]|{[a-z0-9_]*}/i', $expr, $matches);
```

Comment:
rather should rely on Model's callbacks.. OK for now. Or rather Persistence should do this.